### PR TITLE
CSHARP-3268: Update to Cake v1.1.0.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
EG: https://evergreen.mongodb.com/version/609be9099ccd4e52639e23fb

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -1,9 +1,7 @@
-#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0
-#addin nuget:?package=Cake.FileHelpers&version=3.3.0
-#addin nuget:?package=Cake.Git&version=0.22.0
-#addin nuget:?package=Cake.Incubator&version=5.1.0
-#tool dotnet:?package=GitVersion.Tool&version=5.3.7
-#tool nuget:?package=xunit.runner.console
+#addin nuget:?package=Cake.FileHelpers&version=4.0.1
```
Comment (DmitryLukyanov) [inline build.cake:1]: changed versions to the latest

File: build.cake [hunk 2]
```diff
@@ -1,9 +1,7 @@
-#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0
-#addin nuget:?package=Cake.FileHelpers&version=3.3.0
-#addin nuget:?package=Cake.Git&version=0.22.0
-#addin nuget:?package=Cake.Incubator&version=5.1.0
-#tool dotnet:?package=GitVersion.Tool&version=5.3.7
-#tool nuget:?package=xunit.runner.console
```
Comment (DmitryLukyanov) [inline build.cake:6]: as I see we don't use this

File: evergreen/compile.sh [hunk 3]
```diff
@@ -14,10 +14,10 @@ if [ "Windows_NT" = "$OS" ]; then # Magic variable in cygwin
     setx $var z:\\data\\tmp
     export $var=z:\\data\\tmp
   done
-  powershell.exe .\\build.ps1 -target Build -Verbosity Diagnostic
+  powershell.exe .\\build.ps1 --target Build --Verbosity Diagnostic
```
Comment (DmitryLukyanov) [inline evergreen/compile.sh:17]: now, linux and macos versions requires double `--`, for consistency I made the same change for windows as well

File: tools/packages.config [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <packages>
-    <package id="Cake" version="0.38.4" />
+    <package id="Cake" version="1.1.0" />
```
Comment (DmitryLukyanov) [inline tools/packages.config:3]: to test it locally it can be required to reset a Tools folder

**Review by JamesKovacs — APPROVED** [review]
Cake now implicitly bootstraps itself. So we don't need the bootstrap in the final steps of `build.ps1` and `build.sh`.

https://cakebuild.net/blog/2021/02/cake-v1.0.0-released#implicit-bootstrapping

That said, there is no harm in bootstrapping and it keeps our `build.sh` script in sync with:

https://github.com/cake-build/cake/blob/develop/build.sh

LGTM.

**Comment by DmitryLukyanov** [comment]
i wasn't able to reduce a number of cake tools only to:
>#module nuget:?package=Cake.DotNetTool.Module&version=0.4.0
>#tool dotnet:?package=GitVersion.Tool&version=5.6.9
or 
>#tool dotnet:?package=GitVersion.Tool&version=5.6.9

since the error with GitVersion:https://evergreen.mongodb.com/task_log_raw/dot_net_driver_secure_tests__version~2.6_os~windows_64_topology~replicaset_auth~auth_ssl~ssl_test_net452_patch_48b9d2036417d05537a1776bf3bbc78592302b6b_60a05e7456234366c2e509c3_21_05_15_23_51_16/0?type=T#L813
So probably we can leave it as is for now

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**